### ISSUE TITLE
Error running playbooks with collection version >2.9.0

### ISSUE THREAD

**Issue by honvl** [PaloAltoNetworks/pan-os-ansible#351]
## Describe the bug

Running Ansible Tower 3.7.2 with Ansible version 2.9.27. With collection versions >2.9.0, there is a syntax error in panos.py that prevents the running of any panos module. Collection version 2.9.0 works fine. Tested by varying the version in requirements.yml

requirements.yml
---
collections:
  - name: paloaltonetworks.panos
    version: 2.9.0

## Expected behavior

TASK [Palo Alto info] **********************************************************
task path: /tmp/awx_3794_zh3xnp39/project/panosop.yml:6
<fw_px_pa3220-2_mgt.ad.dpw.com> ESTABLISH LOCAL CONNECTION FOR USER: awx
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c 'echo ~awx && sleep 0'
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /var/lib/awx/.ansible/tmp `"&& mkdir "` echo /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143 `" && echo ansible-tmp-1668121427.81-87-210007363038143="` echo /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143 `" ) && sleep 0'
Using module file /tmp/awx_3794_zh3xnp39/requirements_collections/ansible_collections/paloaltonetworks/panos/plugins/modules/panos_op.py
<fw_px_pa3220-2_mgt.ad.dpw.com> PUT /var/lib/awx/.ansible/tmp/ansible-local-3CXiOeH/tmp9QMRoV TO /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143/AnsiballZ_panos_op.py
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c 'chmod u+x /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143/ /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143/AnsiballZ_panos_op.py && sleep 0'
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c '/usr/bin/python /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143/AnsiballZ_panos_op.py && sleep 0'
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c 'rm -f -r /var/lib/awx/.ansible/tmp/ansible-tmp-1668121427.81-87-210007363038143/ > /dev/null 2>&1 && sleep 0'
[DEPRECATION WARNING]: Python library "pandevice" is now "pan-os-python" and is
 now 1.0! Please "pip install pan-os-python" at your earliest convenience.. 
This feature will be removed in version 3.0.0. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: Classic provider params are deprecated; use "provider" 
instead. This feature will be removed in version 3.0.0. Deprecation warnings 
can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [fw_px_pa3220-2_mgt.ad.dpw.com] => {
    "changed": true, 

## Current behavior

TASK [Palo Alto info] **********************************************************
task path: /tmp/awx_3792_1b6a2cjq/project/panosop.yml:6
<fw_px_pa3220-2_mgt.ad.dpw.com> ESTABLISH LOCAL CONNECTION FOR USER: awx
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c 'echo ~awx && sleep 0'
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /var/lib/awx/.ansible/tmp `"&& mkdir "` echo /var/lib/awx/.ansible/tmp/ansible-tmp-1668121298.95-87-181019727973196 `" && echo ansible-tmp-1668121298.95-87-181019727973196="` echo /var/lib/awx/.ansible/tmp/ansible-tmp-1668121298.95-87-181019727973196 `" ) && sleep 0'
<fw_px_pa3220-2_mgt.ad.dpw.com> EXEC /bin/sh -c 'rm -f -r /var/lib/awx/.ansible/tmp/ansible-tmp-1668121298.95-87-181019727973196/ > /dev/null 2>&1 && sleep 0'

"msg": "Unexpected failure during module execution.",
    "exception": "Traceback (most recent call last):\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py\", line 147, in run\n    res = self._execute()\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py\", line 665, in _execute\n    result = self._handler.run(task_vars=variables)\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py\", line 47, in run\n    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py\", line 825, in _execute_module\n    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py\", line 211, in _configure_module\n    **become_kwargs)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 1283, in modify_module\n    environment=environment)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 1120, in _find_module_utils\n    py_module_cache, zf)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 751, in recursive_finder\n    [os.path.join(*py_module_name[:-idx])])\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 671, in __init__\n    self.get_source()\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 686, in get_source\n    pkg = import_module(self._package_name)\n  File \"/usr/lib64/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n  File \"/usr/lib/python2.7/site-packages/ansible/utils/collection_loader.py\", line 107, in load_module\n    mod = self._find_module(fullname, None, load=True)[1]\n  File \"/usr/lib/python2.7/site-packages/ansible/utils/collection_loader.py\", line 221, in _find_module\n    code_object = compile(source=source, filename=source_path, mode='exec', flags=0, dont_inherit=True)\n  File \"/tmp/awx_3792_1b6a2cjq/requirements_collections/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 694\n    **ref_spec,\n              ^\nSyntaxError: invalid syntax\n",

## Steps to reproduce

1.
2.
3.
4.

## Context

Broken playbooks, unable to use playbooks from official Palo Alto ansible-playbooks repository

## Your Environment

- Collection:
- Python: 2.7.5
- Ansible: 2.9.27
- PAN-OS Python Library & version (e.g. pandevice 0.14.0, pan-os-python 1.0.2):

**Comment by shinmog** [comment]
The example playbooks probably need to be updated.  Quite a bit has changed in the last few releases, and they haven't kept up.

Speaking of old, you got output that is informing you about the pandevice repo that was renamed to pan-os-python.  You definitely need to update to pan-os-python, as there is functionality in the current (v2.12.0) collection that makes use of it.

Furthermore, the line of code that error is pointing to is extremely old.  The collection has seen massive changes and improvements in functionality since 2.9.0.  You need to update your collection code as well.  IMPORTANT NOTE:  you want to update to the latest v2.x release, NOT the v3.0.0 release.  The v3.0.0 release is actually v2.10.0, and we've been hounding RedHat to remove that release for months.

TL;DR:

1. update to pan-os-python latest.
2. update the collection to the latest v2.x release.

**Comment by honvl** [comment]
Updated to pan-os-python latest, and
The deprecation warning for pan-os-python has disappeared when running with collection version 2.9.0. 
However, attempting to run with collection 2.12.0 gives a similar error as before, except the ref_spec line number has changed to 707.

(ansible) [root@pyansible1 ~]# pip install pan-os-python
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting pan-os-python
  Downloading https://files.pythonhosted.org/packages/64/38/4192a1f395ffd14858d7c6a308720316441275de530b2f391c4d11cd52a7/pan_os_python-1.7.3-py2.py3-none-any.whl (145kB)
     |â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 153kB 1.0MB/s 
Collecting pan-python<0.18.0,>=0.17.0
  Downloading https://files.pythonhosted.org/packages/e1/00/6b97526c080cb1df314377b4ac39ccd7a7664aced447f912bcbde9bae422/pan_python-0.17.0-py2.py3-none-any.whl (59kB)
     |â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 61kB 5.4MB/s 
Installing collected packages: pan-python, pan-os-python
  Found existing installation: pan-python 0.16.0
    Uninstalling pan-python-0.16.0:
      Successfully uninstalled pan-python-0.16.0
Successfully installed pan-os-python-1.7.3 pan-python-0.17.0
WARNING: You are using pip version 19.3.1; however, version 20.3.4 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.


below output after changing collections.yml to:
name: paloaltonetworks.panos
version: 2.12.0

{
    "msg": "Unexpected failure during module execution.",
    "exception": "Traceback (most recent call last):\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py\", line 147, in run\n    res = self._execute()\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py\", line 665, in _execute\n    result = self._handler.run(task_vars=variables)\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py\", line 47, in run\n    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py\", line 825, in _execute_module\n    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)\n  File \"/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py\", line 211, in _configure_module\n    **become_kwargs)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 1283, in modify_module\n    environment=environment)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 1120, in _find_module_utils\n    py_module_cache, zf)\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 751, in recursive_finder\n    [os.path.join(*py_module_name[:-idx])])\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 671, in __init__\n    self.get_source()\n  File \"/usr/lib/python2.7/site-packages/ansible/executor/module_common.py\", line 686, in get_source\n    pkg = import_module(self._package_name)\n  File \"/usr/lib64/python2.7/importlib/__init__.py\", line 37, in import_module\n    __import__(name)\n  File \"/usr/lib/python2.7/site-packages/ansible/utils/collection_loader.py\", line 107, in load_module\n    mod = self._find_module(fullname, None, load=True)[1]\n  File \"/usr/lib/python2.7/site-packages/ansible/utils/collection_loader.py\", line 221, in _find_module\n    code_object = compile(source=source, filename=source_path, mode='exec', flags=0, dont_inherit=True)\n  File \"/tmp/awx_3821_16dhqvze/requirements_collections/ansible_collections/paloaltonetworks/panos/plugins/module_utils/panos.py\", line 707\n    **ref_spec,\n              ^\nSyntaxError: invalid syntax\n",
    "_ansible_no_log": false,
    "stdout": ""
}

**Comment by shinmog** [comment]
Ok, so line 707 is pointing at a dictionary expansion used within a module that performs vsys importing.  There are a few things that import into a vsys, the main one is network interfaces.

Using the following Ansible playbook, I cannot reproduce the error message you're seeing:

```yaml
- name: Network Playbook
  hosts: fw
  connection: local
  gather_facts: false

  collections:
    - paloaltonetworks.panos

  tasks:
  - name: Grab auth creds
    include_vars: 'vars.yml'
    no_log: 'yes'

  - name: Create and import a ethernet interface
    panos_interface:
      provider: '{{ aws_provider }}'
      if_name: 'ethernet1/4'
      mode: 'layer3'
      vsys: 'vsys1'
      state: 'present'
    register: ans

  - debug:
      msg: '{{ ans }}'
```

Running the playbook for the first time, I get `changed=True`, then `changed=False` the second time.

So, doing a bit of light searching on Google makes me think the dict expansion that the code is doing is syntax that was added in python > 3.5.

In my dev environment, I'm running python 3.7.6 (but an older version of Ansible).

In the CI for this collection, we've seen failures / errors telling us RedHat is going to be setting the lower bound of python to 3.8.

TL;DR - Upgrade to python >= 3.8

**Comment by honvl** [comment]
Ok, thank you for finding the root of the issue. I will consider moving to Ansible Automation Platform 2.2 with Python >= 3.8, which I understand requires a python interpreter line in the inventory.
